snaps/qemu-op-tee-kernel/snapcraft.yaml

```diff
@@ -41,41 +59,155 @@ parts:
+      - LIBDIR: "/usr/lib/aarch64-linux-gnu"
+      - DESTDIR: "${CRAFT_PART_INSTALL}"
+      - PKG_CONFIG: "pkg-config"
+      - PKG_CONFIG_PATH: "/usr/lib/aarch64-linux-gnu/pkgconfig"
+    override-build: |
+      make O=${CRAFT_PART_BUILD}/out -j$(nproc)
+      make install O=${CRAFT_PART_BUILD}/out
+    prime:
+      - usr/lib/*/lib*so*
+
+  optee-uc-fde-client:
+    after:
+      - optee-client
+      - optee-os
+      - optee-uc-fde
+    plugin: nil
+    source-type: local
+    build-packages:
+      - libjson-c-dev:arm64
+      - libcryptsetup-dev:arm64
+    build-environment:
+      - ARCH: "arm64"
+      - CROSS_COMPILE: "aarch64-linux-gnu-"
+      - TA_DEV_KIT_DIR: "${CRAFT_STAGE}/export-ta_arm64"
+      - OPTEE_CLIENT_EXPORT: "${CRAFT_STAGE}/usr"
+      - DESTDIR: "${CRAFT_PART_INSTALL}"
+      - LDFLAGS: "-L${CRAFT_STAGE}/usr/lib/aarch64-linux-gnu"
+    override-build: |
+      make -C ${CRAFT_STAGE}/optee-uc-fde O=${CRAFT_PART_BUILD}/out fde-key-manager fde-reveal-key fde-setup fde-helper
+      make -C ${CRAFT_STAGE}/optee-uc-fde O=${CRAFT_PART_BUILD}/out fde-key-manager fde-reveal-key fde-setup fde-helper install
+      install -D -m 755 ${CRAFT_PART_INSTALL}/usr/bin/fde-setup ${CRAFT_PART_INSTALL}/meta/hooks/fde-setup
+      ln -sf fde-reveal-key ${CRAFT_PART_INSTALL}/usr/bin/fde-setup
+    prime:
+      - meta/
+
+  chroot:
+    plugin: nil
+    build-packages:
+      - wget
       - zstd
     override-build: |
       # building the initrd for an arm64 system requires that we use a arm64
       # chroot with ubuntu-core-initramfs installed in it
-      debootstrap --variant minbase --arch arm64 noble ./fs http://ports.ubuntu.com/ubuntu-ports
-      cat <<EOF > ./fs/etc/apt/sources.list
-      deb http://ports.ubuntu.com/ubuntu-ports noble main restricted universe multiverse
-      deb http://ports.ubuntu.com/ubuntu-ports noble-updates main restricted universe multiverse
-      deb http://ports.ubuntu.com/ubuntu-ports noble-security main restricted universe multiverse
-      EOF
+      wget https://cdimage.ubuntu.com/ubuntu-base/noble/daily/current/noble-base-arm64.tar.gz
+
+      mkdir fs
+      tar -xvf ./noble-base-arm64.tar.gz -C ./fs
+
+      cp --no-dereference /etc/resolv.conf ./fs/etc/resolv.conf
+      for m in proc run sys dev dev/pts dev/null dev/zero dev/full dev/random dev/urandom dev/tty; do
+        mount --bind "/${m}" "./fs/${m}"
+      done
+
       chroot ./fs apt-get update
-      chroot ./fs apt-get install -y software-properties-common
+      chroot ./fs apt-get install -y software-properties-common linux-firmware
       chroot ./fs add-apt-repository ppa:snappy-dev/image -y
       chroot ./fs apt-get update
-      chroot ./fs apt-get install -y ubuntu-core-initramfs linux-firmware
+      chroot ./fs apt-get install -y ubuntu-core-initramfs
 
-      kernel_version=$(basename "${CRAFT_STAGE}/modules/"*)
+      for m in proc run sys dev; do
+        umount -R "./fs/${m}"
+      done
```

Comment:
To make sure things are unmounted even when there are errors you could define this before the mounts:
```
finish() {
    umount ... || true
}
trap finish EXIT
```